src/components/ui/members/password-update-form.tsx

```diff
@@ -0,0 +1,130 @@
+"use client"
+
+import { useState, type FormEvent, type ChangeEvent } from "react"
+import { Button } from "@/components/ui/button"
+import { Input } from "@/components/ui/input"
+import { Label } from "@/components/ui/label"
+import { NewUserPassword } from "@/types/user"
+
+interface PasswordUpdateFormProps {
+    onSubmit: (updatedPassword: NewUserPassword) => Promise<void>
+    isSubmitting: boolean
+}
+
+export function PasswordUpdateForm({ onSubmit, isSubmitting }: PasswordUpdateFormProps) {
+    const [formData, setFormData] = useState<NewUserPassword>({
+        current_password: "",
+        new_password: "",
+        confirm_password: "",
+    })
+    const [errors, setErrors] = useState<Record<string, string>>({})
+
+    const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+        const { name, value } = e.target
+        setFormData((prev) => ({ ...prev, [name]: value }))
+
+        // Clear error when user types
+        if (errors[name]) {
+            setErrors((prev) => {
+                const newErrors = { ...prev }
+                delete newErrors[name]
+                return newErrors
+            })
+        }
+    }
+
+    const validateForm = () => {
```

Comment:
We should also enforce a minimum password length of 8 or more characters which will set us up for more stringent password rules in the future, but I don't think we need to apply those just yet, only minimum length.

Apologies if this feels like a significant enough curveball that we didn't discuss ahead of time. I think it's a significant enough security risk to be able to set an empty or overly short password on a login form that faces the public internet.

Reply:
@calebbourg what about this one?